### DISCUSSION TITLE
libXext.6.dylib not found

### DISCUSSION THREAD

**Discussion by MMcKester** [ImageMagick/ImageMagick#6723] [Contribute]
Hi, I downloaded the latest macOS binaries of ImageMagick and followed the instructions as described on https://imagemagick.com/script/download.php#macosx

I use macOS `11.1 (20C69)` and receive the following error:

`dyld: Library not loaded: /opt/X11/lib/libXext.6.dylib`

Is there any chance to start magick without the X11 dependency? I simply executed `./magick`. The documentation states:

> Note, the display program requires the X11 server available on your Mac OS X installation DVD. Once that is installed, you will also need to set export DISPLAY=:0.

But AFAIK i didn't start the `display` program. Any help is highly appreciated!

Thank you!

**Comment by urban-warrior** [comment]
Until we have a chance to review this issue, see https://imagemagick.org/script/download.php#macosx.

**Comment by MMcKester** [comment]
Thank you! Are there any build instructions available how the homebrew package got build?

**Comment by sanieldalib** [comment]
+1 On this same issue. I currently install from `brew` which works, but using the binary yields this issue.

**Comment by matthewpflueger** [comment]
+1 Same issue - using the MacOS binary fails due to missing X11 libs...

**Comment by aneeshbaby** [comment]
Same issue for me. Because of SSL certificate issue I cannot install using the command `brew install imagemagick` hence tried the binaries and got in the same trouble. 

The error I am getting when try to install using `brew install imagemagick` is;

> zzz$ brew install imagemagick
> Updating Homebrew...
> ==> Auto-updated Homebrew!
> Updated 1 tap (homebrew/core).
> ==> Updated Formulae
> Updated 3 formulae.
> 
> ==> Downloading https://ghcr.io/v2/homebrew/core/libpng/manifests/1.6.37
> 
> curl: (60) SSL certificate problem: unable to get local issuer certificate
> More details here: https://curl.haxx.se/docs/sslcerts.html
> 
> curl failed to verify the legitimacy of the server and therefore could not
> establish a secure connection to it. To learn more about this situation and
> how to fix it, please visit the web page mentioned above.
> Error: Failed to download resource "libpng_bottle_manifest"
> Download failed: https://ghcr.io/v2/homebrew/core/libpng/manifests/1.6.37

By the way, it worked on another computer.

**Comment by doveq** [comment]
> +1 Same issue - using the MacOS binary fails due to missing X11 libs...

me too, macOS Big Sur error:  dyld: Library not loaded: /opt/X11/lib/libXt.6.dylib

sh: 
export MAGICK_HOME="/Users/abc/ImageMagick-7.0.10/" && export DYLD_LIBRARY_PATH="/Users/abc/ImageMagick-7.0.10/lib/" && /Users/abc/ImageMagick-7.0.10/bin/convert

**Comment by gehui1983** [comment]
me too

**Comment by KidA001** [comment]
Same issues. I am unable to use brew so I need to install manually. Getting this same install error.

**Comment by bioworldmerch** [comment]
./configure --help will give you options. You want --without-x as an option. You can also do same in brew to customize the install. you have to google it and figure out how to configure imagemagick install in brew.

**Comment by anewzhuang** [comment]
dyld[11124]: Library not loaded: /opt/X11/lib/libXt.6.dylib
me too...
Is there any solution if i don't want to install with homebrew?

**Comment by cocode** [comment]
I was able to build from source, and it was quite easy. 

Follow the Linux instructions here: https://imagemagick.org/script/install-source.php.  Only takes a copy of commands, and a few minutes to compile. It worked first try, for me. You'll need Xcode, I assume.  Directions for installing that are easily googlable. Seemed to work fine, but I haven't done much testing yet. (make check passed, so I assume things are good).

Another possibility (which I haven't tried) is to install X11 using XQuartz. I *think* that should provide the /opt/X11/lib/libXext.6.dylib library. Then just use the pre-built binary.

I would assume that the problem is the binary was built expecting X11. It's an interesting question whether it's better for the binary distribution to do more, but have more dependencies, or do less, but be easier to install. Maybe have one base version, and one "kitchen sink" binary"?

Personally, I think file conversion and anything to do with display should be in completely different projects, just because of dependency hell. But no one asked me :-)

**Comment by nicolaibach** [comment]
> Another possibility (which I haven't tried) is to install X11 using XQuartz. I _think_ that should provide the /opt/X11/lib/libXext.6.dylib library. Then just use the pre-built binary.

That worked for me on macOS Catalina 10.15.7

**Comment by mattpr** [comment]
```
dyld: Library not loaded: /opt/X11/lib/libXt.6.dylib
  Referenced from: /Users/mattp/Downloads/ImageMagick-7.0.10/./bin/identify
  Reason: image not found
```

I don't know anything about IM's internals, but is X11 really a necessary dependency for `identify`?

Not in a position to judge, but I would think that X11 would only be a dependency when we want IM to "display" something for us.  Per the download page:

> Note, the [display](https://imagemagick.org/script/display.php) program requires the X11 server available on your Mac OS X installation DVD.

**Comment by snibgo** [comment]
On your computer, perhaps `identify` and the other programs are symbolic links to `magick`.

**Reply by mattpr** [reply]
Indeed.  Apparently the official macos release just symlinks everything.

```
ImageMagick-7.0.10 % ls -lha ./bin
total 96
drwxr-xr-x@ 18 mattp  staff   576B Nov 23  2020 .
drwxr-xr-x@  8 mattp  staff   256B Jan  8 20:45 ..
-rwxr-xr-x@  1 mattp  staff   1.4K Nov 23  2020 Magick++-config
-rwxr-xr-x@  1 mattp  staff   1.5K Nov 23  2020 MagickCore-config
-rwxr-xr-x@  1 mattp  staff   1.3K Nov 23  2020 MagickWand-config
lrwxr-xr-x@  1 mattp  staff     6B Nov 23  2020 animate -> magick
lrwxr-xr-x@  1 mattp  staff     6B Nov 23  2020 compare -> magick
lrwxr-xr-x@  1 mattp  staff     6B Nov 23  2020 composite -> magick
lrwxr-xr-x@  1 mattp  staff     6B Nov 23  2020 conjure -> magick
lrwxr-xr-x@  1 mattp  staff     6B Nov 23  2020 convert -> magick
lrwxr-xr-x@  1 mattp  staff     6B Nov 23  2020 display -> magick
lrwxr-xr-x@  1 mattp  staff     6B Nov 23  2020 identify -> magick
lrwxr-xr-x@  1 mattp  staff     6B Nov 23  2020 import -> magick
-rwxr-xr-x@  1 mattp  staff    34K Nov 23  2020 magick
lrwxr-xr-x@  1 mattp  staff     6B Nov 23  2020 magick-script -> magick
lrwxr-xr-x@  1 mattp  staff     6B Nov 23  2020 mogrify -> magick
lrwxr-xr-x@  1 mattp  staff     6B Nov 23  2020 montage -> magick
lrwxr-xr-x@  1 mattp  staff     6B Nov 23  2020 stream -> magick
```

**Comment by snibgo** [comment]
So, if you build IM to use X11 (`./configure with-x`) but the library is not present at run-time, there will be problems.

**Reply by mattpr** [reply]
And apparently this is how the official binary builds are done.

The above was from the [official macos binary release](https://imagemagick.org/script/download.php#macosx) (not built by me).

Not sure the statistics about how many macos users use/want the "display" (X11 required) features of IM.  Definitely not relevant for me as I just want the CLI stuff.  ...but at least the download page readme should be updated to warn that X11 is required for ALL functions for the binary on that page since they all symlink to magick which depends on x11.  Basically no point in downloading that binary release unless you are willing to also setup X11 too.

**Comment by urban-warrior** [comment]
ImageMagick includes the X11 library as an optional dependency. Disabling it means the CLI commands will work fine, except for `magick display`, `magick animate`, and `magick import`, as they specifically rely on X11 for proper functionality.

**Reply by mattpr** [reply]
The official mac binary seems to just symlink everything to a single `magick` bin that REQUIRES (not optional) X11 regardless of what command you run (ie `identify` fails due to missing X11 which makes no sense).

Would be super if maybe there were 2 mac binary releases, one with X11 required and one without.

**Reply by mattpr** [reply]
The current release page for the macos bin says:

> Note, the [display](https://imagemagick.org/script/display.php) program requires the X11 server available on your Mac OS X installation DVD. Once that is installed, you will also need to set export DISPLAY=:0.

Per my [previous post](https://github.com/ImageMagick/ImageMagick/discussions/6723#discussioncomment-8055660), it appears X11 is a required dependency of everything in the official macos bin...ie, running `identify` errors with missing X11...so not just about `display`.

So if the official bin isn't changed to make X11 optional (`display` commands error if X11 not found but everything else works), it would be nice to at least update the README to be more accurate...something along the lines of:

> This official IM macos build REQUIRES X11.  If you don't need `display` related functionality, you can build IM yourself without the X11 dependency.

**Reply by simurgh9** [reply]
> Would be super if maybe there were 2 mac binary releases, one with X11 required and one without.

I think this will be best.

**Comment by tpimh** [comment]
Anyone providing prebuilt IM without X for macOS?

**Reply by tpimh** [reply]
So I couldn't find what I was searching for, so I made it myself. I have built this binary for my CPU and OS, so it might not run on your platform. It has libMagickWand, libMagicCore and libpng compiled-in statically, and is dynamically linked against zlib. Use at your own risk, but it works for me: [magick-with-static-libpng-x86_64-apple-darwin23.3.0.zip](https://github.com/ImageMagick/ImageMagick/files/14213410/magick-with-static-libpng-x86_64-apple-darwin23.3.0.zip)

**Comment by fmw42** [comment]
For display on Mac with Imagemagick, use xQuartz

**Comment by tinng81** [comment]
Can confirm install X11 would provide the dylib. This is using the provided ImageMagick binaries.

```
$ magick --version
dyld: Library not loaded: /opt/X11/lib/libfreetype.6.dylib
  Referenced from: /Users/winng/ImageMagick/bin/magick
  Reason: image not found
zsh: abort      magick --version

$ brew install --cask xquartz
...

$ magick --version
Version: ImageMagick 7.0.10-43 Q16 x86_64 2020-11-23 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI
Delegates (built-in): bzlib freetype jng jpeg lcms png raw tiff webp x xml zlib
```
